### PR TITLE
Add Instagram plugin

### PR DESCRIPTION
**Pull request by MSzabi**
## Type of Change

- **Markdown-It Plugins:** Instagram

## What issue does this relate to?
N/A

### What should this PR do?
This introduces a new plugin with syntax for embedding posts from Instagram in Markdown. The syntax is [instagram <URL>], following the same pattern as many of our other embed plugins.

Like other embed plugins, it also supports optional flags after the URL, in this case for setting the alignment (left/center/right), and the width (between 326 and 550, per Instagram's docs).

### What are the acceptance criteria?
The new syntax works as expected, allowing the user to embed Instagram posts in Markdown content, optionally changing the post alignment or setting a custom width for the post.

### CODE REVIEW

File: rules/embeds/instagram.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        // If silent, don't replace
+        if (silent) return false;
+
+        // Get current string to consider (just current line)
+        const pos = state.bMarks[startLine] + state.tShift[startLine];
+        const max = state.eMarks[startLine];
+        const currentLine = state.src.substring(pos, max);
+
+        // Perform some non-regex checks for speed
+        if (currentLine.length < 13) return false; // [instagram a/p/b]
```
Comment (MattIPv4) [inline rules/embeds/instagram.js:82]:
```suggestion
        if (currentLine.length < 13) return false; // [instagram a]
```

**Review by MattIPv4 — CHANGES_REQUESTED** [review]

File: rules/embeds/instagram.js (hunk 1 above)
Comment (MSzabi) [reply rules/embeds/instagram.js:82]: Changed.

File: rules/embeds/instagram.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+/**
+ * Add support for [Instagram](https://instagram.com/) embeds in Markdown, as block syntax.
+ *
+ * The basic syntax is `[instagram <post>]`. E.g. `[instagram https://www.instagram.com/p/CkQuv3_LRgS]`.
+ * After the post, assorted space-separated flags can be added (in any combination/order):
+ *
+ * - Add `caption` to include caption under the post.
+ * - Add `left`, `center`, or `right` to set the alignment of the embed (default is `left`).
+ * - Add any set of digits to set the width of the embed (in pixels, between 326 and 550, default is 540).
```
[the remark this answers is not in the document]
Comment (MSzabi) [reply rules/embeds/instagram.js:33]: Yeah, initially I have defaulted from code to 540, then realized it's way too big, so I left the default that was coming from instagram. Forgot to update it here.
Done.

File: styles/_instagram.scss [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+limitations under the License.
+*/
+
+.instagram {
+    .instagram-media {
+        margin: 1rem auto !important;
+    }
+
+    &[align="left"] {
+        .instagram-media{
```
[the remark this answers is not in the document]
Comment (MSzabi) [reply styles/_instagram.scss:23]: Fixed.

File: rules/embeds/instagram.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        if (state.inlineMode) return;
+
+        // Check if we need to inject the script
+        if (state.env.instagram && state.env.instagram.tokenized && !state.env.instagram.injected) {
+            // Set that we've injected it
+            state.env.instagram.injected = true;
+
+            // Inject the token
+            const token = new state.Token('html_block', '', 0);
+            token.content = '<script async defer src="//www.instagram.com/embed.js" type="text/javascript"></script>\n';
```
[the remark this answers is not in the document]
Comment (MSzabi) [reply rules/embeds/instagram.js:144]: Updated.

File: rules/embeds/instagram.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+ * <div class="instagram" align="left">
+ *     <blockquote class="instagram-media"
+ *       style="width: 400px;"
+ *       data-instgrm-permalink="https://www.instagram.com/p/CkQuv3_LRgS"
+ *       data-instgrm-version="14"
+ *       data-instgrm-captioned>
+ *          <a href="https://instagram.com/p/CkQuv3_LRgS">View post</a>
+ *     </blockquote>
+ * </div>
+ * <script async defer src="//www.instagram.com/embed.js" type="text/javascript"></script>
```
[the remark this answers is not in the document]
Comment (MSzabi) [reply rules/embeds/instagram.js]: Introduced the alignment flag after the initial write-up and forgot to update it here. Should match now.

File: rules/embeds/instagram.js [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+/**
+ * Add support for [Instagram](https://instagram.com/) embeds in Markdown, as block syntax.
+ *
+ * The basic syntax is `[instagram <post>]`. E.g. `[instagram https://www.instagram.com/p/CkQuv3_LRgS]`.
+ * After the post, assorted space-separated flags can be added (in any combination/order):
+ *
+ * - Add `caption` to include caption under the post.
+ * - Add `left`, `center`, or `right` to set the alignment of the embed (default is `left`).
+ * - Add any set of digits to set the width of the embed (in pixels, between 326 and 550, default is 326 not enforced here, but provided by Instagram's embed.js).
```
Comment (MattIPv4) [inline rules/embeds/instagram.js:33]:
```suggestion
 * - Add any set of digits to set the width of the embed (in pixels, between 326 and 550, default is 326 as set by Instagram's embed.js).
```
Just cleaning up the wording here (please do the same to the README)

**Review by MattIPv4 — CHANGES_REQUESTED** [review]

**Review by MattIPv4 — CHANGES_REQUESTED** [review]

File: fixtures/full-input.md [hunk 7]
```diff
@@ -341,6 +341,17 @@ Like a few other embeds, you can also pass optional flags to customize the embed
...
 
+### Instagram
+
+You can also embed a post from Instagram by passing the URL for the post:
+
+[instagram https://www.instagram.com/p/CkQuv3_LRgS]
+
+Like a few other embeds, you can also pass optional flags to customize the embed:
+
+- Pass any integer value (between 250 and 550) to set a custom width for the embed (e.g. `[instagram https://www.instagram.com/p/CkQuv3_LRgS 400]`)
```
[the remark this answers is not in the document]
Comment (MSzabi) [reply fixtures/full-input.md:352]: Updated.

File: fixtures/full-output.html [hunk 8]
```diff
@@ -343,6 +343,18 @@ <h3 id="twitter"><a class="hash-anchor" href="#twitter" aria-hidden="true"></a>T
...
+<h3 id="instagram"><a class="hash-anchor" href="#instagram" aria-hidden="true"></a>Instagram</h3>
+<p>You can also embed a post from Instagram by passing the URL for the post:</p>
+<div class="instagram">
+    <blockquote class="instagram-media" data-instgrm-permalink="https://www.instagram.com/p/CkQuv3_LRgS" data-instgrm-version="14">
+        <a href="https://instagram.com/p/CkQuv3_LRgS">View post</a>
+    </blockquote>
+</div>
+<p>Like a few other embeds, you can also pass optional flags to customize the embed:</p>
+<ul>
+<li>Pass any integer value (between 250 and 550) to set a custom width for the embed (e.g. <code>[instagram https://www.instagram.com/p/CkQuv3_LRgS 400]</code>)</li>
```
[the remark this answers is not in the document]
Comment (MSzabi) [reply fixtures/full-output.html:355]: Updated.

File: fixtures/full-input.md [hunk 9]
```diff
@@ -341,6 +341,17 @@ Like a few other embeds, you can also pass optional flags to customize the embed
...
+- Pass `caption` to include caption under the post (e.g. `[instagram https://www.instagram.com/p/CkQuv3_LRgS caption]`)
```
[the remark this answers is not in the document]
Comment (MSzabi) [reply fixtures/full-input.md:353]: Me sorry for not adding it in the first place...
Fixed.

File: rules/embeds/instagram.js [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+        if (state.inlineMode) return;
+
+        // Check if we need to inject the script
+        if (state.env.instagram && state.env.instagram.tokenized && !state.env.instagram.injected) {
+            // Set that we've injected it
+            state.env.instagram.injected = true;
+
+            // Inject the token
+            const token = new state.Token('html_block', '', 0);
+            token.content = '<script async defer src="https://www.instagram.com/embed.js" type="text/javascript"></script>\n';
```
[the remark this answers is not in the document]
Comment (MSzabi) [reply rules/embeds/instagram.js:144]: Nice catch. Fixed.

**Review by MattIPv4 — APPROVED** [review]
LGTM

**Pull request merged**